src/TableauConnector.js

```diff
@@ -122,8 +123,8 @@ class TableauConnector {
     tableau.authType = tableau.authTypeEnum.custom
 
     this.authenticate()
-      .then(accessToken => this.validateAccessIfNeeded(accessToken))
-      .then(accessToken => {
+      .then(({accessToken, refreshToken}) => this.validateAccessIfNeeded(accessToken, refreshToken))
```

Comment:
`accessToken` is being passed through this chain, but it isn't needed anywhere. I'd suggest passing `refreshToken` only.